docs/src/DeveloperDocumentation/new_developers.md

```diff
@@ -295,3 +295,133 @@ git fetch oscar-system
 git rebase oscar-system/master
 ```
 Adding the remote only has to be executed once.
+
+### Pitfalls: Mutable objects in Oscar code
+
+The objects in Oscar that can be changed (mutable objects) can be linked
+together into an arbitrary dependency graph. Therefore, a change to one object
+may have unintended consequences on another object in the system.
+The simplest way to avoid such unintended behaviour is to *never* modify an
+existing object. However, if you intended to use mutable objects the way they
+were intended to be used, namely, for mutation, then Oscar provides plenty of
+places to burn your fingers.
+
+The simplest example is the creation of a polynomial ring. If we mutate the
+array of symbols used for printing, we have effectively changed the ring.
+
+```
+julia> v = [:x, :y, :z]; R = PolynomialRing(QQ, v)[1]
+Multivariate Polynomial Ring in x, y, z over Rational Field
+
+julia> v[3] = :w; R
+Multivariate Polynomial Ring in x, y, w over Rational Field
+```
+
+It turns out that strings do not have this effect.
+
+```
+julia> v = ["x", "y", "z"]; R = PolynomialRing(QQ, v)[1]
+Multivariate Polynomial Ring in x, y, z over Rational Field
+
+julia> v[3] = "w"; R
+Multivariate Polynomial Ring in x, y, z over Rational Field
+```
+
+This behaviour where changes to one object can have hidden and unexpected
+effects on another object is by no means limited to polynomial rings.
+Here we create the factored element `x = 2^3`. By modifying the object `2`, we
+are able to change `x` as well.
+
+```julia
+julia> a = fmpz(2)
+2
+
+julia> x = FacElem([a], [fmpz(3)]); evaluate(x)
+8
+
+julia> a = one!(a)  # inplace assignment of a to 1
+1
+
+julia> evaluate(x)  # x has been changed
+1
+```
+
+In the previous example, the link between the object `x` and the object `a` can
+be broken by passing a `deepcopy` of `a` to the `FacElem` function.
+
+```julia
+julia> a = fmpz(2)
+2
+
+julia> x = FacElem([deepcopy(a)], [fmpz(3)]); evaluate(x)
+8
+
+julia> a = one!(a)  # modyfy a
+1
+
+julia> evaluate(x)  # x is now unchanged
+8
+```
+
+The previous two examples dealt with the fact that ownership conventions on
+function *arguments* are completely unspecified in Oscar. If you are going to
```

Comment:
I don't find this style of writing very constructive or productive. You are right to point out an issue, but how about us working to rectify it by trying to formulate some rules how ownership for arguments and return values should work by default? Then we could document that, and start to fix functions that violate the rules (by changing their behavior, or perhaps in some cases, by changing their documentation to explicitly point out the deviation).

Also, once again, I am fairly certain this problem is not unique to Oscar within the Julia realm, or in general.

Reply:
The new version might be slightly more helpful in this regard. I point out that ownership conventions are an implementation detail with some small guarantees. The default is therefore that you cannot expect to own anything after passing it around.

Reply:
Thank you